input_group: Add InputGroup component.

sxhxliang opened the pull request:
- Introduce `InputGroup` component with flexible layout and alignment options
- Add supporting components: `InputGroupAddon`, `InputGroupText`, `InputGroupInput`, and `InputGroupTextarea`
- Implement various examples including search, URL input, validation, and chat interfaces
- Update story module to include and initialize InputGroupStory
- Register InputGroup in the main UI module exports


## Description

Add a new [shadcn/ui InputGroup component]( https://ui.shadcn.com/docs/components/input-group) that provides a flexible way to compose input fields with addons such as icons, buttons, and text labels.  

### Features:

  - InputGroup: Main container with support for disabled/invalid states, horizontal/vertical layouts
  - InputGroupAddon: Flexible addon container with alignment options (InlineStart, InlineEnd, BlockStart,
   BlockEnd)
  - InputGroupText: Simple text element for labels within addons
  - InputGroupInput: Styled input wrapper that integrates seamlessly with the group
  - InputGroupTextarea: Multi-line text input variant for chat-like interfaces

### Examples included:

  - Basic search with icon and results count
  - URL input with protocol prefix
  - Username with validation badge
  - Search with clear button and loading state
  - Chat input with bottom toolbar
  
## Screenshot

| Before                       | After                       |
| ---------------------------- | --------------------------- |
| Windows/- |<img width="928" height="626" alt="image" src="https://github.com/user-attachments/assets/7a735e99-916b-493c-bd7c-e0b830e74bfe" />|
| macOS/- |<img width="2562" height="1652" alt="image" src="https://github.com/user-attachments/assets/5e85ebbd-b633-421a-90af-84b4bf924752" />|


## Break Changes

Describe any breaking changes introduced by this pull request. If none, remove this section.

- Change 1

None

## How to Test

  1. Run the story application:
  cargo run
  2. Navigate to the InputGroup story panel
  3. Verify the following examples render correctly:
    - Basic Search with Icon and Results
    - URL Input with Protocol Prefix
    - Username with Validation Icon
    - Search with Clear Button
    - Search with Loading State
    - Multiple Icons
    - With Action Button
    - Disabled State
    - Invalid/Error State
    - Chat Input with Toolbar

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) document and followed the guidelines.
- [x] Reviewed the changes in this PR and confirmed AI generated code (If any) is accurate.
- [x] Passed `cargo run` for story tests related to the changes.
- [x] Tested macOS, Windows  platforms performance (if the change is platform-specific)

huacnlee commented on crates/ui/src/input/input_group.rs:
```diff
@@ -0,0 +1,461 @@
+    fn default() -> Self {
+        Self::new()
+    }
+}
```
Remove default for InputGroup, just use `new` method.

sxhxliang commented:
## Review Comments Addressed

Thank you for the detailed review! I've addressed all the comments:

### ✅ Completed (12 items)
1. Simplified examples - removed all helper functions, inlined directly in render()
2. Removed section comments (`// === Types ===`, etc.)
3. Removed `DEFAULT_INPUT_GROUP_ID` constant and custom id
4. Removed `invalid` state and related API
5. Removed `Default` trait implementation
6. Fixed disabled style - now uses `muted` background + overlay to block all interactions (mouse, keyboard, scroll)
7. Added helper methods: `align()`, `block_end()`, `block_start()`, `inline_end()`, `inline_start()` in alphabetical order
8. Removed `InputGroupText::label()` method
9. Simplified types: `Entity<InputState>` instead of full paths
10. Removed `InputGroupTextarea::height()` method
11. Updated all doc examples to reflect API changes
12. Removed `cursor_text()` from addon to avoid wrong cursor affordance

### ⚠️ Two Items Need Discussion

**1. Padding methods (line 241-252)**
Current implementation uses individual `pl/pr/pt/pb` because different alignments need different padding values. A single `p()` method cannot handle tuples. Is this acceptable, or would you prefer a different approach?

**2. min_w_0() (line 135)**  
The linter removed this, but it's needed to prevent flex overflow. Should I restore it with a comment, or handle differently?

All changes have been tested and compilation verified.

huacnlee commented:
Sorry, I have consider about this PR more time. But I still not sure this is should be accept.

Because the Input is already have `prefix`, `sufixx` can support things like InputGroup. The example of InputGroup is pretty like the Input example.

So I just close this, untill when I have new mind.
